src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -34,7 +34,15 @@ public async Task<ImmutableArray<DiagnosticData>> ForceAnalyzeProjectAsync(Proje
                 if (!_projectToForceAnalysisData.TryGetValue(project, out var box))
                 {
                     box = new(await ComputeForceAnalyzeProjectAsync().ConfigureAwait(false));
+
+                    // Try to add the new computed data to the CWT.  But use any existing value that another thread
+                    // might have beaten us to storing in it.
+#if NET
+                    _projectToForceAnalysisData.TryAdd(project, box);
+                    Contract.ThrowIfFalse(_projectToForceAnalysisData.TryGetValue(project, out box));
```

Comment:
can be replaced with GetOrAdd https://github.com/dotnet/runtime/pull/111204 when we move to NET10